onnxruntime/core/language_interop_ops/torch/torch_proxy.cc

```diff
@@ -0,0 +1,361 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/language_interop_ops/torch/torch_proxy.h"
+
+#include <Python.h>
+#include "core/dlpack/dlpack_converter.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/language_interop_ops/torch/custom_function_register.h"
+#include "core/language_interop_ops/torch/object_pointer.h"
+#include "core/language_interop_ops/torch/refcount_tracker.h"
+#include "core/language_interop_ops/torch/gil.h"
+#include "core/platform/env.h"
+
+namespace onnxruntime {
+namespace language_interop_ops {
+namespace torch {
+
+template <>
+void ObjectPointer<PyObject>::free() {
+  Py_XDECREF(ptr);
+}
+
+#ifndef NDEBUG
```

Comment:
I like including the debugging checks and extra diagnostics here!  [ Think this explains why I did not see the output -- I have only been testing with RelWithDebInfo ]

Suggest moving the #if..#else..#endif inside the function though so that they are properly nested.  We can use ORT_UNUSED_PARAMETER within the non-debug branch to handle log_tag.

Reply:
Thanks for the suggestion. I just planned to do that. :)

Reply:
I'll move all MACRO inside RefCountTracker, then all other references don't need care about build flags.